Cuis-University-COOP.pck.st

```diff
@@ -1098,6 +1286,70 @@ withSelector: messageName applyCondition: aMessageNode
 	
 	or: [ aMessageNode isMessage: messageName receiver:  self literalZero arguments: self messageSize] .! !
 
+!COOPRuleNonsenseBoolean methodsFor: 'printing' stamp: 'MEG 10/26/2019 16:54:05'!
+description
+
+	^ RuleDescriptor new descriptionForNonsenseBoolean! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'printing' stamp: 'MEG 10/26/2019 16:46:29'!
+title
+	^ RuleDescriptor new titleForNonsenseBoolean .! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'testing' stamp: 'MEG 10/26/2019 17:06:09'!
+applyCondition: aMessageNode
+
+ 	^ self booleanControllingMessages anySatisfy: 
+		[ :controllingMessage | self meetCondition: controllingMessage on: aMessageNode ]! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'testing' stamp: 'MEG 10/26/2019 17:11:33'!
+booleanOnArguments: arguments
+
+	^ arguments anySatisfy: 
+		[ :argument | self statementWithBooleanVariableNode: argument statements ]! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'testing' stamp: 'MEG 10/26/2019 15:06:10'!
+canHandle: aNode 
+	
+	^ aNode isMessageNode! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'testing' stamp: 'MEG 10/26/2019 17:05:16'!
+keyIsABoolean: key
+
+	^ key = 'true' or: [ key = 'false' ] ! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'testing' stamp: 'MEG 10/26/2019 17:11:33'!
+statementWithBooleanVariableNode: statements
+ 
+	^ statements anySatisfy: 
+		[ :statement | ( statement isVariableNode ) 
+			and: [ self keyIsABoolean: statement key ] ]! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'matcher' stamp: 'MEG 10/26/2019 17:06:09'!
+meetCondition: messageName on: messageNode 
+	
+	^ self withSelector: messageName searchOnArguments: messageNode! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'matcher' stamp: 'MEG 10/26/2019 17:10:31'!
+withSelector: messageName searchOnArguments: messageNode
+	
+	^ ( messageNode isMessageNamed: messageName ) and: [ self booleanOnArguments: messageNode arguments ]! !
+
+!COOPRuleNonsenseBoolean methodsFor: 'information' stamp: 'MEG 10/26/2019 17:08:13'!
+booleanControllingMessages
+	
+	| booleanControllingMessages |
+	
+	booleanControllingMessages _ Set new.
```

Comment:
Te recomiendo acá que hagas una variable de instancia, y lo metas en el initialize.
Podes aplicar un refactor que se llama _Temporary to Instance Variable_, y luego mover lo demás al initialize.
¿Por qué? Porque creamos un set cada vez que vamos a analizar un método, no está bueno, mata la performance aunque no lo creas.

Reply:
Excelente idea! :100: